### PR TITLE
Change Construction Kit to GECK

### PR DESCRIPTION
**Pull request by Twinki14**
# Motivations
GECK is a more accurate name for the Fallout specific creation kit

# Modifications
- Rename Construction Kit to GECK, which is more well known

### CODE REVIEW

**Comment by Silarn** [comment]
I'll go with the general consensus on this but I personally feel like having the support isn't a bad thing. There are definitely going to be old school people still using these things and the fact that we display the application if it exists isn't telling a new user to go download them. The only way it shows up is if someone already installed it.

As far as Creation Kit vs GECK, I'm ambivalent. I mean 'Creation Kit' is in the name so I don't see it as being that different or confusing. It's also a more generally correct name across all Beth games as GECK is a Fallout specific name. But most people still understand 'creation kit' regardless of context.

**Comment by Twinki14** [comment]
> I'll go with the general consensus on this but I personally feel like having the support isn't a bad thing. There are definitely going to be old school people still using these things and the fact that we display the application if it exists isn't telling a new user to go download them. The only way it shows up is if someone already installed it.
> 
> As far as Creation Kit vs GECK, I'm ambivalent. I mean 'Creation Kit' is in the name so I don't see it as being that different or confusing.

I agree, but that's generally why we have versioning; If there are old-schoolers out there, they can use older versions of the plugin or add the executables manually. For newer modders, these extra options can be generally confusing, and in my opinion should be simply removed.

We aren't losing anything here with this PR.

**Comment by Silarn** [comment]
Well the thing is MO2 plugins are fairly tied to the MO2 version. Any changes to the plugin ABI are going to break older builds, leaving anyone that wants the support but wants an updated MO2 out in the cold. I think we're kind of bad about updating plugin versions in general, but that's a different issue.

**Comment by Twinki14** [comment]
Fair point, but couldn't old schoolers still add their FOMM or BOSS executables manually? MO2 will still be supporting that

**Comment by Holt59** [comment]
Regarding the list of executables, I don't really see any positive point in removing the old ones. If the executables cannot be found, they will not be displayed anyway. For GECK, I don't really mind.

Regarding the plugins, what would be the benefits of marking them as primary? If we do so, you'll need to remove them from `DLCPlugins`. I'm personally not really for this since I like seeing my DLCs as DLCs, but that might be subjective.

If we decide to go with the changes, you'll need to fix the formatting issues.

**Comment by Silarn** [comment]
Yeah I didn't see or fully consider the DLC change. While I understand the intent, I don't think it's necessarily appropriate to cater only to people who are just buying the game new. It's still entirely possible to have an old copy of NV / FO3 / etc. and these plugins are, as far as I know, still not hard coded to load with the game in a specific order.

Flagging DLC plugins doesn't just indicate that they're installed as DLC but it has implications about how they can be loaded or ordered in the plugin list. Generally primary plugins should be ones which the game loads automatically in a specific order and will cause it to fail if they don't exist.

File: src/gamefalloutnv.cpp [hunk 1]
```diff
@@ -149,9 +149,7 @@ QList<ExecutableInfo> GameFalloutNV::executables() const
   ExecutableInfo launcher("Fallout Launcher", findInGameFolder(getLauncherName()));
   QList<ExecutableInfo> extraExecutables =
       QList<ExecutableInfo>()
-      << ExecutableInfo("Fallout Mod Manager", findInGameFolder("fomm/fomm.exe"))
-      << ExecutableInfo("Construction Kit", findInGameFolder("geck.exe"))
-      << ExecutableInfo("BOSS", findInGameFolder("BOSS/BOSS.exe"))
+      << ExecutableInfo("GECK", findInGameFolder("geck.exe"))
```
Comment (Silarn) [inline src/gamefalloutnv.cpp]: You can keep the 'GECK' name change I suppose, but I think we'd prefer to keep the other executables for legacy reasons. They will only populate if these applications were already installed by the user so there's no clear reason to remove a convenience layer here.

File: src/gamefalloutnv.cpp [hunk 2]
```diff
@@ -246,7 +244,11 @@ QString GameFalloutNV::steamAPPId() const
 
 QStringList GameFalloutNV::primaryPlugins() const
 {
-  return {"falloutnv.esm"};
+  return {
```
Comment (Silarn) [inline src/gamefalloutnv.cpp]: I think we're going to remove this for now. There are both technical and legacy reasons to leave this alone and there are use cases for FNV players to allow them to toggle and reorder the DLCs if they want to. TTW is a different matter.

**Review by Silarn — CHANGES_REQUESTED** [review]

**Comment by Twinki14** [comment]
PR has been re-focused into just the GECK name change

**Review by Holt59 — APPROVED** [review]

**Review by Silarn — APPROVED** [review]

**Pull request merged**